tests/client/function_run.go

```diff
@@ -215,13 +215,34 @@ func (c *Client) WaitForRunStatus(
 	return run
 }
 
-// retrieve run with traces
-// TODO: add the traces once implemented
-func (c *Client) RunTraces(ctx context.Context, runID string) *RunV2 {
+func (c *Client) WaitForRunTracesWithTimeout(ctx context.Context, t *testing.T, runID *string, status models.FunctionStatus, timeout time.Duration, interval time.Duration) *RunV2 {
+	var traces *RunV2
+	require.NotNil(t, runID)
+	require.Eventually(t, func() bool {
+		if *runID == "" {
+			return false
+		}
+
+		run, err := c.RunTraces(ctx, *runID)
```

Comment:
Here we simply keep retrying until we see no more errors, return data are present, and the status matches.